src/Entity/User.php

```diff
@@ -4,73 +4,86 @@
 
 use ApiPlatform\Core\Annotation\ApiResource;
 use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\DiscriminatorColumn;
+use Doctrine\ORM\Mapping\DiscriminatorMap;
+use Doctrine\ORM\Mapping\InheritanceType;
 
 /**
  * @ApiResource()
  * @ORM\Entity(repositoryClass="App\Repository\UserRepository")
+ * @InheritanceType("JOINED")
+ * @DiscriminatorColumn(name="discr", type="string")
+ * @DiscriminatorMap({"student" = "Student", "lector" = "Lector"})
```

Comment:
`  * @DiscriminatorMap({"student" = "Student", "lector" = "Lector"})` na
`  * @DiscriminatorMap({"student": "Student", "lector": "Lector"})`

v dokumentácii to majú trochu blbo... áno, funguje to, ale takto je to lepšie :D